### DISCUSSION TITLE
Video

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#299] [Component suggestions]
<img width="480" alt="visuele weergave van de video component" src="https://github.com/user-attachments/assets/18e62563-cb94-4049-83fe-3b726b65b062">

We willen graag meer leren over de Video component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1729594801160569)

---

## Naam
Video

## Doel
Video met bedieningselementen voor afspelen, pauzeren, tekstalternatieven, en meer.

## Ook bekend als
Video Player, Media Player

## Gerelateerde componenten
[Youtube Video](https://github.com/orgs/nl-design-system/discussions/354), [Figure](https://github.com/orgs/nl-design-system/discussions/221)

## Onderzoek

[Accessible multimedia - MDN docs](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/Multimedia)
Artikel waarin wordt beschreven dat het native HTML video element niet toegankelijk genoeg is en hoe je hier textuele alternatieve voor kan aanbieden.

[Browser video players review - Adrian Roselli](https://adrianroselli.com/2023/09/browser-video-players-review.html)
Adrian Roselli test de toegankelijkheid van het native HTML `<video>` element. Dat doet hij onder andere voor toetsenbord, schermlezer, stembediening en 'forced colors'. Sowieso heeft Adrian veel geschreven over [Video en toegankelijkheid](https://adrianroselli.com/?s=Video)

[Accessible HTML5 Media Players & Resources - Digita11y](https://www.digitala11y.com/accessible-jquery-html5-media-players/)
Verzameling van toegankelijke mediaplayers en richtlijnen.

## Notities

Deze notities van @Robbert zijn overgezet uit uit Backlog issue:

- De video player moet een tekstalternatief kunnen hebben in expandable region, bijvoorbeeld met een transcriptie.
- De labels in de user interface moeten beschikbaar zijn in de taal van de pagina, dus de player moet een mogelijkheid hebben om UI labels te vertalen.
- De labels van de buttons moeten passen bij de actie. Bij toggle buttons moeten `aria-label` aangepast worden wanneer ze pressed zijn:
  - Play/pauze: "Afspelen" vs "Pauzeer"
  - Fullscreen: "Schermvullende weergave openen"
  - Ondertitels: "Ondertiteling aan" vs "Ondertiteling uit"
  - Descriptive audio track: "Audiobeschrijving afspelen" vs "Audiobeschrijving stoppen"
  - Video description: "Gebarentaal tonen" vs "Gebarentaal verbergen"
- Toggle buttons moeten `aria-pressed` hebben, bijvoorbeeld:
- Onderzoeken of pressed toggle buttons met een aangepast label duidelijk zijn:
  - toggle button, pressed, Pauzeer
  - toggle button, pressed, Ondertiteling uit
  - toggle button, pressed, Audiobeschrijving stoppen
  - toggle button, pressed, Gebarentaal verbergen
- Onderzoeken of tooltips nodig zijn voor icon buttons, in plaats van `title` attribuut.
- Vermijd de title attribuut, daarvan kan de tooltip over de video heen staan zonder dat de button focus heeft.
- Keyboard shortcuts zonder modifier keys mogen alleen werken als iets de bijbehorende button focus heeft. Bijvoorbeeld `M` als de mute button focus heeft. Naar aanleiding van dit artikel: [Keyboard shortcuts need modifier keys](https://hidde.blog/keyboard-shortcuts/)
- Icon buttons moeten minimaal 24x24 of 48x48 zijn, afhankelijk van WCAG versie
- Spatie button moet wel toggle buttons bedienen, niet video pauzeren (`stopPropagation()`).
- Spatie button: op welke plekken mag de spatie button de video pauzeren? Overal in de video, of alleen wanneer de play button focus heeft? Hoe zit het in full screen modus?
- Hoe wordt de video duration beschikbaar voor screen reader gebruikers? Duration moet een misschien duidelijke onzichtbare description hebben: "Videoduur: 2:30". Misschien `aria-describedby`?
- Tekst description moet paragraphs gebruiken, zodat je kunt skippen of terugspoelen met screen readers.
- Video: moet de video in een `figure` staan?
- Moet video een titel hebben?
- Moet titel van de video in een heading of in een `figcaption` staan?
- Is het nodig dat de beschrijvende tekst direct na de video komt, voor de download links?
- Link in de documentatie naar een voorbeeld van toegankelijke download links.
- Icon buttons met zichtbare tekst moeten de tekst in het label hebben. Bijvoorbeeld: `aria-label="Closed Captions (CC)"`.

Deze notities van @jeffreylauwers zijn overgezet uit uit Backlog issue:
- Niet enkel op kleur vertrouwen om de status van de video controls te communiceren.
- Moeten de 'controls' in beeld blijven bij full screen?
- Moeten de 'controls' een zichtbaar label hebben met het oog op spraakbediening?

**Comment by jeffreylauwers** [comment]
## Amsterdam

[Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29204)

---

### Screenshots / Varianten

**Figma**

Muis uit
![Scherm­afbeelding 2024-10-21 om 11 49 33](https://github.com/user-attachments/assets/76f8765b-99b9-480c-812b-cb14e07fc125)

Muis in
![Scherm­afbeelding 2024-10-21 om 11 49 38](https://github.com/user-attachments/assets/78d85aee-486e-479c-8577-983a5e2e7eb5)

'Player' met opvolgend: Pauze, Pauze en Replay
![Scherm­afbeelding 2024-10-21 om 11 50 31](https://github.com/user-attachments/assets/f56faea2-c920-4753-9079-bfd37ce0d3b0)

Voorbeeld met componenten Button en Detail als onderdeel van de Player
![Scherm­afbeelding 2024-10-21 om 11 50 03](https://github.com/user-attachments/assets/6ef8b97e-e8a2-4756-9045-a1f6b6765c0f)

---

### Notities

- Gevonden in een Branch met label 'Todo'.
- Er wordt gesproken over 'Muis uit' en 'Muis in'. Mijn vermoeden is dat dit interactie rondom 'hover' betreft.
- Opties beschikbaar voor:
  - Play, Pauze, Replay
  - Geluid: aan/uit
  - Tijdsindicatie: huidig + totaal
  - Progress: totaal, buffer, huidig + knob.
  - Ondertiteling: aan/uit
  - Volledig scherm: aan/uit
- Onder de video worden de opties voor 'Download' en 'Uitgeschreven tekst' aangeboden.

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/video)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-10-21 om 12 05 25](https://github.com/user-attachments/assets/96695fad-88c9-4d5e-bf29-86b9fa10e21f)

Collapse 'Download deze video' expanded
![Scherm­afbeelding 2024-10-21 om 12 05 34](https://github.com/user-attachments/assets/63a04517-0805-45a4-95dd-6953e1d2b80e)

Collapse 'Uitgeschreven tekst' expanded
![Scherm­afbeelding 2024-10-21 om 12 05 44](https://github.com/user-attachments/assets/d0b0846e-6b4a-48e6-bc82-6103291764da)

---

### Notities

- Documentatie beschikbaar.
- Opties beschikbaar voor:
  - Play, Pauze
  - Geluid: aan/uit + volumeregelaar
  - Tijdsindicatie: huidig + totaal
  - Progress: totaal + knob.
  - Ondertiteling: aan/uit
  - Volledig scherm: aan/uit
- Onder de video worden opties voor 'Download deze video' en 'Uitgeschreven tekst' aangeboden.
- Implementatie is op basis van https://plyr.io/

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

- [Documentatie website](https://componenten.lenteveld.nl/mediaplayer)
- [GitHub - AccessibleMediaPlayer](https://github.com/OpenGemeenten/AccessibleMediaPlayer)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-10-21 om 14 40 27](https://github.com/user-attachments/assets/e054bc1a-7a2b-4ead-a59d-625d462cd622)

Tekstalternatief - Expanded
![Scherm­afbeelding 2024-10-21 om 14 40 44](https://github.com/user-attachments/assets/93428715-8817-4bd8-ba66-6453702a6555)


---

### Notities

- Documentatie beschikbaar voor gebruik op een toegankelijke manier.
- Opties beschikbaar voor:
  - Play, Pauze, Replay
  - Tijdsindicatie: huidig + totaal
  - Progress: totaal, huidig, buffer + knob (bij hover).
  - Geluid: aan/uit + volumeregelaar
  - Ondertiteling: aan/uit
  - Audi descriptie: aan/uit
  - Gebarentaal: aan/uit
  - Volledig scherm: aan/uit
- Onder de video wordt een optie voor 'Tekstalternatief' aangeboden.

**Reply by martijnrietveld** [reply]
Je kunt in bovenstaande mediaplayer ook een link naar Youtube of Vimeo toevoegen. In dat geval wordt er automatisch toestemming voor externe bron gevraagd. Bij een externe bron kun je geen audiobeschrijving of gebarentaal toevoegen. Door deze oplossing is het niet meer nodig om op de homepage al toestemming te vragen voor cookies
<img width="732" alt="toestemming externebron" src="https://github.com/user-attachments/assets/f7441ec6-0be5-44ac-b3cc-8142dcd687f9">

**Reply by martijnrietveld** [reply]
https://componenten.lenteveld.nl/mediaplayer

**Comment by jeffreylauwers** [comment]
## ICTU

[Github: ICTU - Digitale Overheid WP - rijksvideoplugin](https://github.com/ICTU/ictuwp-plugin-rijksvideo)

Videoplayer door [Paul van Buuren](https://github.com/paulvanbuuren)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Video op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=video)
